graph_train.py

```diff
@@ -1,183 +0,0 @@
-        """
-        This function is called after every epoch.
-        In this function, the dev accuracy is checked and the learning rate is adjusted if dev accuracy has decreased
-        """
-        val_acc = trainer.callback_metrics.get('val_acc').item()
-        print('Epoch [' + str(trainer.current_epoch + 1) + ']; train_loss: ' + str(
-            trainer.callback_metrics.get('train_loss').item())
-              + '; val_loss: ' + str(trainer.callback_metrics.get('val_loss').item()) + ': val_acc: ' + str(val_acc))
-
-
-def train_model(args):
-    os.makedirs(args.log_dir, exist_ok=True)
-
-    if args.dataset == 'r8':
-        dataset = R8Graph(device)
-    elif args.dataset == 'r52':
-        dataset = R52Graph(device)
-
-    # data = data_object.data.to(device)
-    graph_data_loader = geom_data.DataLoader(dataset, batch_size=1)
-
-    # Create a PyTorch Lightning trainer with the generation callback
-    gen_callback = GenerateCallback()
-    trainer = pl.Trainer(default_root_dir=args.log_dir,
-                         checkpoint_callback=ModelCheckpoint(save_weights_only=True, mode="min", monitor="val_loss"),
-                         gpus=1 if torch.cuda.is_available() else 0,
-                         max_epochs=args.epochs,
-                         callbacks=[gen_callback],
-                         progress_bar_refresh_rate=1 if args.progress_bar else 0)
-    trainer.logger._default_hp_metric = None
-
-    # Create model
-    pl.seed_everything(args.seed)
-    model = Graph_model(len(dataset.iton))
-
-    # optionally resume from a checkpoint
-    if args.resume != None:
-        if os.path.isfile(args.resume):
-            print('=> loading checkpoint %s' % args.resume)
-            model = NLI_model.load_from_checkpoint(args.resume)
-            print('=> loaded checkpoint %s' % args.resume)
-        else:
-            print('=> no checkpoint found at %s' % args.resume)
-
-    # Training
-    trainer.fit(model, graph_data_loader, graph_data_loader)
-
-
-if __name__ == '__main__':
-    # Feel free to add more argument parameters
-    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-
-    # Model hyperparameters
-    parser.add_argument('--dataset', default='r8', type=str,
-                        help='What dataset to use',
-                        choices=['r8', 'r52'])
-    parser.add_argument('--embedding_dim', default=300, type=int,
-                        help='Embedding dimensionalities')
-    parser.add_argument('--classifier_hidden', default=512, type=int,
-                        help='Hidden dimensionalities to use inside the classifier')
-    parser.add_argument('--LSTM_hidden', default=2048, type=int,
-                        help='Hidden dimensionalities to use inside the LSTM model')
-    parser.add_argument('--output_dim', default=3, type=int,
-                        help='Number of classes')
-    parser.add_argument('--resume', default=None, type=str, metavar='PATH',
```

Comment:
Would be nice to keep this parameter to be able to resume from a checkpoint. The other arguments can be thrown away.